Source/BSGConnectivity.m

```diff
@@ -25,8 +25,57 @@
 //
 
 #import "BSGConnectivity.h"
+#import "Bugsnag.h"
 
-typedef void (^CallbackBlock)(SCNetworkReachabilityFlags flags);
+static SCNetworkReachabilityRef bsg_reachability_ref;
+BSGConnectivityChangeBlock bsg_reachability_change_block;
+SCNetworkReachabilityFlags bsg_current_reachability_state;
+
+/**
+ * Check whether the connectivity change should be noted or ignored.
+ *
+ * @return YES if the connectivity change should be reported
+ */
+BOOL BSGConnectivityShouldReportChange(SCNetworkReachabilityFlags flags) {
+    #if TARGET_OS_TV || TARGET_OS_IPHONE
+        // kSCNetworkReachabilityFlagsIsWWAN does not exist on macOS
+        const SCNetworkReachabilityFlags importantFlags = kSCNetworkReachabilityFlagsIsWWAN | kSCNetworkReachabilityFlagsReachable;
+    #else
+        const SCNetworkReachabilityFlags importantFlags = kSCNetworkReachabilityFlagsReachable;
+    #endif
+    __block BOOL shouldReport = YES;
+    // Check if the reported state is different from the last known state (if any)
+    if ((flags & importantFlags) != (bsg_current_reachability_state & importantFlags)) {
+        // When first subscribing to be notified of changes, the callback is
+        // invoked immmediately even if nothing has changed. So this block
+        // ignores the very first check, reporting all others.
+        if (bsg_current_reachability_state == 0) {
+            shouldReport = NO;
+        }
+        // Cache the reachability state to report the previous value representation
+        bsg_current_reachability_state = flags & kSCNetworkReachabilityFlagsReachable;
+        #if TARGET_OS_TV || TARGET_OS_IPHONE
+            bsg_current_reachability_state |= flags & kSCNetworkReachabilityFlagsIsWWAN;
+        #endif
+    } else {
+        shouldReport = NO;
+    }
+    return shouldReport;
+}
+
+/**
+ * Textual representation of a connection type
+ */
+NSString *BSGConnectivityFlagRepresentation(SCNetworkReachabilityFlags flags) {
+    BOOL connected = (flags & kSCNetworkReachabilityFlagsReachable);
+    #if TARGET_OS_TV || TARGET_OS_IPHONE
+        return connected
+            ? ((flags & kSCNetworkReachabilityFlagsIsWWAN) ? @"cellular" : @"wifi")
+            : @"none";
+    #else
+        return connected ? @"wifi" : @"none";
+    #endif
+}
```

Comment:
Should the values be represented by constants?  e.g. "wifi" is repeated and so potentially corruptible.  Might also help clarity in tests.